### PR TITLE
Allow zero content types on delete. fixes #69

### LINKED ISSUE #69: Permit zero content types for empty response
The instances for `HasClient` match on the type-level cons eg.

```haskell
instance HasClient (Delete (ct ': cts) ())
```

For empty responses, there isn't a need for any content types; in fact, `ct` and `cts` aren't used in the instance body.

### PR DESCRIPTION
**Pull request by codedmart**
Let me know if anything needs changed.

### CODE REVIEW

**Comment by jkarni** [comment]
Thanks, this looks good. I'll fix the (unrelated) issues on travis before merging, though.

**Comment by jkarni** [comment]
Oh, and could you add an entry to the changelog?

**Comment by codedmart** [comment]
My bad sorry. Of course I will.

File: servant-client/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+0.4.1
+-----
+* `Delete` allow zero `'[]` or more `'[JSON]` content types on delete
```
Comment (christian-marie) [inline servant-client/CHANGELOG.md]: This is a little strangely worded. I find it confusing.

Do you mean something like:

  The `HasClient` instance for `Delete cts ()` now does not care at all about content types provided.

Comment (codedmart) [inline servant-client/CHANGELOG.md]: Sure that sounds fine I will change it.

File: servant-client/src/Servant/Client.hs [hunk 2]
```diff
@@ -123,8 +124,8 @@ instance
 #if MIN_VERSION_base(4,8,0)
          {-# OVERLAPPABLE #-}
 #endif
-  (MimeUnrender ct a) => HasClient (Delete (ct ': cts) a) where
-  type Client (Delete (ct ': cts) a) = EitherT ServantError IO a
+  (MimeUnrender ct a, cts' ~ (ct ': cts)) => HasClient (Delete cts' a) where
```
Comment (christian-marie) [inline servant-client/src/Servant/Client.hs]: Not immediately obvious what's going on here, a comment would be nice.

Comment (codedmart) [inline servant-client/src/Servant/Client.hs]: I will add notes. Thanks!

**Comment by jkarni** [comment]
You might also need to rebase against master to get soenke's travis fixes.

**Comment by alpmestan** [comment]
Would you mind rebasing this against master to get the travis fixes (among other things) ?

**Comment by codedmart** [comment]
@alpmestan no problem. I will do that tomorrow morning. Thanks!

**Comment by codedmart** [comment]
@alpmestan I will need to fix this PR. My rebase and squash went south so there is more in this then there should be. Let me clean it up.

**Comment by codedmart** [comment]
@alpmestan OK now this should be cleaned up. Let me know if I missed anything.

**Comment by codedmart** [comment]
Ok still failing.

**Comment by alpmestan** [comment]
It fails because... `lens-4.10` doesn't build. Interesting. Their [latest travis build](https://travis-ci.org/ekmett/lens) went fine.

**Comment by alpmestan** [comment]
`lens` build failure tracked [here](https://github.com/ekmett/lens/issues/544).

**Comment by codedmart** [comment]
What would you like me to do? Anything?

**Comment by alpmestan** [comment]
Nope. We're busy writing a paper this week but we'll take care of making travis happy and merging your PR this weekend or early next week! Thanks again.

**Comment by jkarni** [comment]
Thanks!

**Pull request merged**